OSX: Getting the following error: call to 'count_digits' is ambiguous

Please let me know if a fix if available?

```
XXX/lib/spdlog/include/spdlog/details/pattern_formatter.h:656:35: error: call to 'count_digits' is ambiguous
        const size_t field_size = fmt::internal::count_digits(msg.thread_id);
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
XXX/lib/spdlog/include/spdlog/fmt/bundled/format.h:904:17: note: candidate function
inline unsigned count_digits(uint64_t n) {
                ^
XXX/lib/spdlog/include/spdlog/fmt/bundled/format.h:1009:17: note: candidate function
inline unsigned count_digits(uint32_t n) {
                ^
In file included from XXX/lib/spdlog/include/spdlog/spdlog.h:13:
In file included from XXX/lib/spdlog/include/spdlog/details/registry.h:15:
In file included from XXX/lib/spdlog/include/spdlog/logger.h:23:
In file included from XXX/lib/spdlog/include/spdlog/sinks/sink.h:9:
In file included from XXX/lib/spdlog/include/spdlog/details/pattern_formatter.h:8:
XXX/lib/spdlog/include/spdlog/details/fmt_helper.h:75:19: error: call to 'count_digits' is ambiguous
    auto digits = fmt::internal::count_digits(n);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
XXX/lib/spdlog/include/spdlog/details/fmt_helper.h:94:5: note: in instantiation of function template specialization 'spdlog::details::fmt_helper::pad_uint<unsigned long, 500>' requested here
    pad_uint(n, 6, dest);
    ^
XXX/lib/spdlog/include/spdlog/details/pattern_formatter.h:472:21: note: in instantiation of function template specialization 'spdlog::details::fmt_helper::pad6<unsigned long, 500>' requested here
        fmt_helper::pad6(static_cast<size_t>(micros.count()), dest);
                    ^
XXX/lib/spdlog/include/spdlog/fmt/bundled/format.h:904:17: note: candidate function
inline unsigned count_digits(uint64_t n) {
                ^
XXX/lib/spdlog/include/spdlog/fmt/bundled/format.h:1009:17: note: candidate function
inline unsigned count_digits(uint32_t n) {
                ^
In file included from XXX/lib/spdlog/include/spdlog/spdlog.h:13:
In file included from XXX/lib/spdlog/include/spdlog/details/registry.h:15:
In file included from XXX/lib/spdlog/include/spdlog/logger.h:23:
In file included from XXX/lib/spdlog/include/spdlog/sinks/sink.h:9:
In file included from XXX/lib/spdlog/include/spdlog/details/pattern_formatter.h:8:
/Users/vadimmelnicuk/Developer/c/hazel/lib/spdlog/include/spdlog/details/fmt_helper.h:100:5: error: no matching function for call to 'pad_uint'
    pad_uint(n, 9, dest);
    ^~~~~~~~
XXX/lib/spdlog/include/spdlog/details/pattern_formatter.h:489:21: note: in instantiation of function template specialization 'spdlog::details::fmt_helper::pad9<unsigned long, 500>' requested here
        fmt_helper::pad9(static_cast<size_t>(ns.count()), dest);
                    ^
XXX/lib/spdlog/include/spdlog/details/fmt_helper.h:72:13: note: candidate template ignored: substitution failure [with T = unsigned long, Buffer_Size = 500]
inline void pad_uint(T n, unsigned int width, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
```

--- a/include/spdlog/details/fmt_helper.h
+++ b/include/spdlog/details/fmt_helper.h
@@ -18,6 +18,7 @@ inline spdlog::string_view_t to_string_view(const fmt::basic_memory_buffer<char,
 {
     return spdlog::string_view_t(buf.data(), buf.size());
 }
+
 template<size_t Buffer_Size1, size_t Buffer_Size2>
 inline void append_buf(const fmt::basic_memory_buffer<char, Buffer_Size1> &buf, fmt::basic_memory_buffer<char, Buffer_Size2> &dest)
 {
@@ -40,6 +41,13 @@ inline void append_int(T n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 {
     fmt::format_int i(n);
     dest.append(i.data(), i.data() + i.size());
+}
+
+template<typename T>
+inline unsigned count_digits(T n)
+{
+    using count_type = std::conditional<(sizeof(std::size_t) > sizeof(std::uint32_t)), std::uint64_t, std::uint32_t>::type;
+    return fmt::internal::count_digits(static_cast<count_type>(n));
 }
 
 template<size_t Buffer_Size>
@@ -69,7 +77,7 @@ template<typename T, size_t Buffer_Size>
 inline void pad_uint(T n, unsigned int width, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 {
     static_assert(std::is_unsigned<T>::value, "append_uint must get unsigned T");
-    auto digits = fmt::internal::count_digits(static_cast<uint64_t>(n));
+    auto digits = count_digits(n);
     if (width > digits)
     {
         const char *zeroes = "0000000000000000000";

--- a/include/spdlog/details/fmt_helper.h
+++ b/include/spdlog/details/fmt_helper.h
@@ -42,7 +42,6 @@ inline void append_int(T n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
     dest.append(i.data(), i.data() + i.size());
 }
 
-
 template<size_t Buffer_Size>
 inline void pad2(int n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 {
@@ -60,23 +59,21 @@ inline void pad2(int n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
         dest.push_back('0');
         dest.push_back(static_cast<char>('0' + n));
     }
-    else    // negatives (unlikely, but just in case, let fmt deal with it)
+    else // negatives (unlikely, but just in case, let fmt deal with it)
     {
         fmt::format_to(dest, "{:02}", n);
     }
 }
 
-
-
 template<typename T, size_t Buffer_Size>
 inline void pad_uint(T n, unsigned int width, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 {
     static_assert(std::is_unsigned<T>::value, "append_uint must get unsigned T");
-    auto digits = fmt::internal::count_digits(n);
-    if(width > digits)
+    auto digits = fmt::internal::count_digits(static_cast<uint64_t>(n));
+    if (width > digits)
     {
-        const char* zeroes = "0000000000000000000";
-        dest.append(zeroes, zeroes + width-digits);
+        const char *zeroes = "0000000000000000000";
+        dest.append(zeroes, zeroes + width - digits);
     }
     append_int(n, dest);
 }
@@ -86,7 +83,6 @@ inline void pad3(T n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 {
     pad_uint(n, 3, dest);
 }
-
 
 template<typename T, size_t Buffer_Size>
 inline void pad6(T n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
@@ -99,8 +95,6 @@ inline void pad9(T n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 {
     pad_uint(n, 9, dest);
 }
-
-
 
 // return fraction of a second of the given time_point.
 // e.g.

--- a/include/spdlog/details/fmt_helper.h
+++ b/include/spdlog/details/fmt_helper.h
@@ -46,7 +46,7 @@ inline void append_int(T n, fmt::basic_memory_buffer<char, Buffer_Size> &dest)
 template<typename T>
 inline unsigned count_digits(T n)
 {
-    using count_type = std::conditional<(sizeof(std::size_t) > sizeof(std::uint32_t)), std::uint64_t, std::uint32_t>::type;
+    using count_type = std::conditional<(sizeof(T) > sizeof(std::uint32_t)), std::uint64_t, std::uint32_t>::type;
     return fmt::internal::count_digits(static_cast<count_type>(n));
 }
 

--- a/include/spdlog/details/pattern_formatter.h
+++ b/include/spdlog/details/pattern_formatter.h
@@ -652,7 +652,7 @@ public:
 
     void format(const details::log_msg &msg, const std::tm &, fmt::memory_buffer &dest) override
     {
-        const auto field_size = fmt::internal::count_digits(static_cast<uint64_t>(msg.thread_id));
+        const auto field_size = fmt_helper::count_digits(msg.thread_id);
         scoped_pad p(field_size, padinfo_, dest);
         fmt_helper::append_int(msg.thread_id, dest);
     }

--- a/include/spdlog/details/pattern_formatter.h
+++ b/include/spdlog/details/pattern_formatter.h
@@ -487,7 +487,6 @@ public:
 
         auto ns = fmt_helper::time_fraction<std::chrono::nanoseconds>(msg.time);
         fmt_helper::pad9(static_cast<size_t>(ns.count()), dest);
-
     }
 };
 
@@ -653,7 +652,7 @@ public:
 
     void format(const details::log_msg &msg, const std::tm &, fmt::memory_buffer &dest) override
     {
-        const size_t field_size = fmt::internal::count_digits(msg.thread_id);
+        const auto field_size = fmt::internal::count_digits(static_cast<uint64_t>(msg.thread_id));
         scoped_pad p(field_size, padinfo_, dest);
         fmt_helper::append_int(msg.thread_id, dest);
     }

